pkg/chain/ethereum/chain.go

```diff
@@ -35,13 +38,12 @@ var (
 
 // EthereumChain is an implementation of ethereum blockchain interface.
 type EthereumChain struct {
-	config                         *ethereum.Config
-	accountKey                     *keystore.Key
-	client                         bind.ContractBackend
-	bondedECDSAKeepFactoryContract *contract.BondedECDSAKeepFactory
-	blockCounter                   *blockcounter.EthereumBlockCounter
-	miningWaiter                   *ethutil.MiningWaiter
-	nonceManager                   *ethutil.NonceManager
+	config       *ethereum.Config
+	accountKey   *keystore.Key
+	client       bind.ContractBackend
+	blockCounter *blockcounter.EthereumBlockCounter
+	miningWaiter *ethutil.MiningWaiter
+	nonceManager *ethutil.NonceManager
```

Comment:
What if we keep `bondedECDSAKeepFactoryContract` and add `fullyBackedBondedECDSAKeepFactoryContract`?
We will initialize them depending on the config file and if not needed leave `nil`.

```suggestion
	config       *ethereum.Config
	accountKey   *keystore.Key
	client       bind.ContractBackend
	blockCounter *blockcounter.EthereumBlockCounter
	miningWaiter *ethutil.MiningWaiter
	nonceManager *ethutil.NonceManager
	bondedECDSAKeepFactoryContract *contract.BondedECDSAKeepFactory
	fullyBackedBondedECDSAKeepFactoryContract *contract.FullyBackedECDSAKeepFactory
```